tools/firtool/firtool.cpp

```diff
@@ -336,6 +342,38 @@ struct EmitSplitHGLDDPass
   }
 };
 
+/// Wrapper pass to dump IR.
+struct DumpIRPass
+    : public PassWrapper<DumpIRPass, OperationPass<mlir::ModuleOp>> {
+  DumpIRPass(const std::string &outputFile)
+      : PassWrapper<DumpIRPass, OperationPass<mlir::ModuleOp>>() {
+    this->outputFile.setValue(outputFile);
+  }
+
+  DumpIRPass(const DumpIRPass &other) : PassWrapper(other) {
+    outputFile.setValue(other.outputFile.getValue());
+  }
```

Comment:
I'm all for re-use here, but I may be missing something. What is the benefit to using a small pass versus a helper function?

Reply:
It's necessary to use a pass because it's not possible to get IR after LowerToHW outside of pass pipeline. To use helper function we have to separate pass pipeline but I don't think it's better.

Reply:
Ah I see, makes sense to me. It's a little weird to see a pass definition inside a tool cpp file... it seems fine for this use case, but I was wondering.